### PR TITLE
feature :  spinner widget

### LINKED ISSUE #1871: new spinner widget
### Is your proposal related to a problem?
Spinners are used to show the loading state of a component or page
We don't have a generic spinner widget to indicate loading.
### Describe the solution you'd like

include spinner widget  in editor
size , colour editing options can be included

### PR DESCRIPTION
**Pull request by stepinfwd**
closes #1871

### CODE REVIEW

**Comment by RicoRobinson** [comment]
Is the committed svg used anywhere?

**Comment by stepinfwd** [comment]
> Is the committed svg used anywhere?

This icon is not  used anywhere else , if you are looking to build new widget icon gets automatically  detected if you use same name as the component in smallcase

**Comment by RicoRobinson** [comment]
> Collaborator

Then never mind my comment, I'll look into ToolJet some more :')

File: frontend/src/Editor/Components/Spinner.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React, { useState, useEffect } from 'react';
+
+export const Spinner = ({ styles, height, darkMode }) => {
+    const { colour, size, } = styles;
+
+    const spinnerBaseStyle = {
+        height,
+        backgroundColor: darkMode ? "#47505D" : "#ffffff",
+        display: 'flex',
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Components/Spinner.jsx:9]: Can we move all static styles to `theme.scss`?

File: frontend/src/Editor/Components/components.js [hunk 2]
```diff
@@ -1665,4 +1665,49 @@ export const componentTypes = [
       },
     },
   },
+  {
+    name: 'Spinner',
+    displayName: 'Spinner',
+    description: 'Spinner can be used as a loader',
```
Comment (Navaneeth-pk) [inline frontend/src/Editor/Components/components.js:1671]: "Spinner can be used to display loading status"

**Comment by Navaneeth-pk** [comment]
image
The background should be transparent.

**Comment by stepinfwd** [comment]
fixed lint issues @Navaneeth-pk

**Review by Navaneeth-pk — APPROVED** [review]

**Pull request merged**